---
title: "This is an example script to filter a VCF"
subtitle: "This is part of a larger best practices project with github"
author: "Nicholas F. Brazeau"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output:
  rmarkdown::html_document:
    theme: lumen
    highlight: monochrome
    toc: true
editor_options: 
  chunk_output_type: console
---


```{r imports, include=F, warning=F, results="hide"}

# -----------------------------------
# The purpose of this script is to filter an example vcf 
# Filtering a VCF has many different approaches/theories (i.e. this is no "recipe" or "truly correct way") -- this is an example and is in no way comprehensive
# See https://gatkforums.broadinstitute.org/gatk/discussion/2806/howto-apply-hard-filters-to-a-call-set
# -----------------------------------


## This is where we import libraries and functions
library(tidyverse)
library(RColorBrewer)
library(grid)
library(DT)
library(vcfR)
source("00_functions.R")

```

```{r read_in_file, include=F, warning=F, results="hide"}
vcffile <- "~/Desktop/polyIBD_Temp/play.vcf"
vcf <- vcfR::read.vcfR(vcffile)
```
# Introduction
This is the VCF filter document for Project <NAME>. Thus far, I have done the following: 

1. Aligned with `bwa-mem` with default parameeters
2. 
3. 

_In addition, the following regions have been excluded from the VCF_: 
1. 
2. 
3. 



Overall, this VCF contains `r nrow(vcf@gt)` variants across `r (ncol(vcf@gt)-1)` samples.

The goal of this script is to filter this VCF following [GATK best practices hard filtering](http://gatkforums.broadinstitute.org/gatk/discussion/2806/howto-apply-hard-filters-to-a-call-set). Throughout this script we will  present summary statistics and plots for various `INFO` and `GT` field parameters that we plan on using for "Hard Filters" (see here for [VCF Specifications](https://samtools.github.io/hts-specs/VCFv4.2.pdf)).  


```{r subset_to_snps, include=F, warning=F, results="hide"}
vcf <-vcfR::extract.indels(vcf, return.indels = F) # subset to SNPs
vcf <- vcf[vcfR::is.biallelic(vcf)] # subset to biallelic SNPs
```
Overall, this VCF contains `r nrow(vcf@gt)` biallelic SNPs across `r (ncol(vcf@gt)-1)` samples.


```{r extract_info_field_df, include=F, warning=F, results="hide"}
infodf <- vcf_info_field_to_tidy(vcf)
infodf_summary <- infodf %>% 
  tidyr::gather(key="INFO_field_param", val="val", 2:ncol(infodf)) %>% 
  dplyr::group_by(INFO_field_param) %>% 
  dplyr::summarise(Loci=n(),
                   Missing = paste0(sum(is.na(val))/Loci,"%"),
                   Min=min(val),
                   Q10 = quantile(val, prob=0.1),
                   Q25 = quantile(val, prob=0.25),
                   Median = median(val),
                   Mean = mean(val),
                   Q75 = quantile(val, prob=0.75),
                   Q90 = quantile(val, prob=0.9),
                   Max = max(val),
                   StDev = sd(val),
                   MAD = mad(val)
)







```


`r DT::datatable(segsite, extensions = 'Buttons', options = list(dom = 'Bfrtip',buttons = c('csv')))`



```{r extract_info_field_df, include=F, warning=F, fig.width=9, fig.width=6}


infodf %>% 
  tidyr::gather(key="INFO_field_param", val="val", 2:ncol(infodf)) %>%
  ggplot(aes(x=INFO_field_param, y=val)) + 
  geom_violin(aes(fill=factor(INFO_field_param))) +
  geom_boxplot(width=0.1) +
  geom_jitter(colour="#bdbdbd", height = 0, width = 0.01, alpha=0.1) +
  facet_wrap(~INFO_field_param, scales = "free") +
  ggtitle("Distribution of INFO Field Parameters") + 
  xlab("INFO Field Parameters") + ylab("Values") + 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(), 
        panel.background = element_blank(),
        panel.border = element_rect(color = "black", fill = NA, size = 0.5), 
        plot.title =  element_text(size=14, family = "Arial", face="bold", hjust=0.5, vjust=0.5),
        axis.ticks = element_blank(),
        axis.text.y = element_text(size=9, family = "Arial",  face="bold"),
        axis.title.y = element_text(size=13, family = "Arial", face="bold"),
        axis.text.x = element_blank(),
        axis.title.x = element_text(size=13, family = "Arial", face="bold"),
        strip.text.x = element_text(size =12, face="bold", family = "Arial"),
        legend.position = "none"
  )




grid::grid.newpage()
grid::grid.draw(rbind(ggplotGrob(plotsnps), ggplotGrob(plotIBD), size = "last"))

```